src/components/search-unavailable-alert/tests/SearchUnavailableAlert.test.tsx

```diff
@@ -0,0 +1,26 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { SearchUnavailableAlert } from '../index';
+import { messages } from '../SearchUnavailableAlert';
+
+describe('<SearchUnavailableAlert />', () => {
+  it('renders with expected content', () => {
+    render(
+      <IntlProvider locale="en">
+        <SearchUnavailableAlert />
+      </IntlProvider>,
+    );
+    Object.entries(messages).forEach(([, message]: [string, { defaultMessage: string }]) => {
+      expect(screen.getByText((message as { defaultMessage: string }).defaultMessage)).toBeInTheDocument();
+    });
```

Comment:
nit: Interesting pattern, haven't seen this done before. What happens if/when not all messages are rendered all at once? For example, if there was a conditionally rendered message in the component? I'm wondering if it would still make sense to be explicit about which messages specifically are expected here.